addOns/sstiscanner/src/main/java/org/zaproxy/zap/extension/sstiscanner/SSTIScanner.java

```diff
@@ -0,0 +1,501 @@
+     * @param msg a request only copy of the original message (the response isn't copied)
+     * @param paramName the parameter name that need to be exploited
+     * @param value the original parameter value
+     */
+    public void efficientScan(HttpMessage msg, String paramName, String value) {
+
+        // The efficient scanner detects the existence of vulnerabilities by causing
+        // and detecting errors. To detect errors we start by looking to how the
+        // responses change when we send different inputs. Later, with this information
+        // we can detect which changes caused by our inputs are not normal and which
+        // may indicate an error.
+        // To detect the normal changes we send an random input. This input has the same
+        // size as the polyglot to detect errors caused by size limits.
+        String alphabet;
+        if (value != null && value.length() > 0) {
+            alphabet = value;
+        } else {
+            alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+        }
+
+        String referenceValue = randomStringFromAlphabet(alphabet, error_polyglots[0].length());
+        HttpMessage refMsg = getNewMsg();
+        setParameter(refMsg, paramName, referenceValue);
+        try {
+            sendAndReceive(refMsg, false);
+        } catch (IOException e) {
+            log.warn(
+                    "SSTI vulnerability check failed for parameter ["
+                            + paramName
+                            + "] due to an I/O error",
+                    e);
+            return;
+        }
+
+        // The input point will have associated the several locations where the user
+        // input is reflected("sinks"). This are the locations where it is possible to
+        // see the result of a successful SSTI. The objects representing this "sink"
+        // points contain information about the usual behavior the server has to
+        // different inputs.
+        InputPoint inputPoint =
+                createInputPointWithAllSinks(
+                        getBaseMsg(), value, refMsg, referenceValue, paramName);
+        inputPoint.addReferenceReqToAllSinkPoints(refMsg, paramName, referenceValue);
+
+        if (isStop()) {
+            return;
+        }
+
+        if (hasSuspectBehaviourWithPolyglot(paramName, inputPoint)) {
+            searchForMathsExecution(paramName, inputPoint, false);
+        }
+        return;
+    }
+
+    private InputPoint createInputPointWithAllSinks(
+            HttpMessage originalMsg,
+            String originalValue,
+            HttpMessage referenceMsg,
+            String referenceValue,
+            String paramName) {
+
+        InputPoint inputPoint = new InputPoint();
+
+        inputPoint.addSinkPoint(
+                new ReflectedSinkPoint(originalMsg, originalValue, referenceMsg, referenceValue));
+
+        Set<Integer> sinks = PersistentXSSUtils.getSinksIdsForSource(originalMsg, paramName);
```

Comment:
Does leveraging the PersistentXSSUtils (and other classes? I haven't noticed so far... oops) mean this is subject to this problem: https://github.com/zaproxy/zaproxy/issues/4656 ?

Reply:
This is the only place where I am using PersistentXSSUtils. Yes it is subject that problem. I am working in other plugin to solve that issue. https://github.com/zaproxy/zaproxy/issues/4656